File: scala/grade-school/src/main/scala/GradeSchool.scala
class School {
  type DB = Map[Int, Seq[String]]
  def add(name: String, g: Int) = {
      db(g) = grade(g)+: name
      // how do i acces db
  }
  def db: DB = Map.empty
  def grade(g: Int): Seq[String] = db.getOrElse(g,Nil)
  def sorted: DB = ???
}

File: python/grade-school/grade_school.py
class School:
    
    def __init__(self):
        self.students = dict()

    def add_student(self, name, grade):
        if grade in self.students:
            self.students[grade].append(name)
            return  self.students
        else :
            self.students[grade] = [name]


    def roster(self):
        all = set ()
        for e in self.students:
            all = all.union(set(self.students[e]))
        all = sorted(all)
        return all


    def grade(self, grade_number):
        return self.students[grade_number]

    def added(self):
        pass
